pkg/deploy/strategy/recreate/recreate.go

```diff
@@ -104,10 +104,9 @@ func (s *RecreateDeploymentStrategy) DeployWithAcceptor(from *kapi.ReplicationCo
 		}
 	}
 
-	// If an UpdateAcceptor is provided and we're trying to scale up to more
-	// than one replica, scale up to 1 and validate the replica, aborting if the
-	// replica isn't acceptable.
-	if updateAcceptor != nil && desiredReplicas > 1 {
+	// If an UpdateAcceptor is provided, scale up to 1 and validate the replica,
+	// aborting if the replica isn't acceptable.
+	if updateAcceptor != nil {
```

Comment:
Good question. The upstream rolling updater rejects desired replicas of 0 as invalid. The pre hook would still execute before we pass it off to the upstream library.

For recreate, we could continue to execute pre/post with 0 desired and just consider the deployment complete, but then the user would get different behavior when using the rolling strategy.

If we're going to allow 0, then we'll need to make a case for changing it upstream as well. Otherwise, we should probably reject 0 for recreate to remain consistent. Can you think of a use case where the user would intend to roll out a deployment with 0 desired?

cc @abhgupta